app/services/sync_provider_from_find.rb

```diff
@@ -162,4 +157,26 @@ def derive
 
     class InvalidFindStatusDescriptionError < StandardError; end
   end
+
+  class CourseStudyModes
+    def initialize(course)
+      @course = course
+    end
+
+    def derive
+      both_modes = %w[full_time part_time]
+      return both_modes if @course.both_study_modes_available?
+
+      from_existing_course_options = @course.course_options.pluck(:study_mode).uniq
```

Comment:
could you just do something like `return (from_existing_course_options + [@course.study_mode]).uniq` at this point?

Reply:
Much nicer, updated!